.github/workflows/trivy.yml

```diff
@@ -0,0 +1,30 @@
+name: Trivy scan
+on: [ pull_request ]
+
+jobs:
+  appsec-trivy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK
+        uses: actions/setup-java@v2
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+
+      - name: Construct docker image name and tag
+        id: image-name
+        run: echo ::set-output name=name::${{ github.event.repository.name }}:$(git rev-parse --short HEAD)
```

Comment:
Generating a tag seems unnecessary for this workflow since the image isn't pushed. You could use a fixed tag (or no tag at all) for the purposes of this scan.

Reply:
The image isn't pushed, but both the build step and scanning step need the same image name, and this is less hacky then reading the output of the build log.